### PR TITLE
boards/wemos-zero: fix Arduio I/O Map

### PR DESCRIPTION
**Pull request by maribu**
### Contribution description

Two pins were incorrectly mapped: D2 and D4 are swapped compared to the official Arduino Zero. In addition the is slightly extended.

#### Pinout for Arduino Zero

Official Pinout

#### Pinout for Clones

Clone Pinout

### Issues/PRs references

None

### CODE REVIEW

File: boards/wemos-zero/doc.txt [hunk 1]
```diff
@@ -6,13 +6,14 @@
 ## Overview
 
 The WeMos Zero is a cheap clone of Arduino Zero featuring a ATSAMD21G18A.
+Theses boards either have a silkscreen "WeMos" or "HW-910" label on their back.
```
Comment (mguetschow) [inline boards/wemos-zero/doc.txt:9]:
```suggestion
These boards either have a silkscreen "WeMos" or "HW-910" label on their back.
```

File: boards/wemos-zero/doc.txt [hunk 2]
```diff
@@ -6,13 +6,14 @@
...
 Unlike the original Zero, it does not include an embedded debugger/programmer.
 Instead, the [UF2 Bootloader](https://github.com/microsoft/uf2-samdx1) is
 used to program this board. The SAMD21 is a ARM Cortex-M0+ micro-controller.
 It has 256KiB of flash memory and 32KiB of RAM.
 
-Note: The official Arduino Zero using the native USB port and the Adafruit
-Metro M0 Express should also work with board definition.
+@note The official Arduino Zero using the native USB port and the Adafruit
+      Metro M0 Express should also work with board definition.
```
Comment (mguetschow) [inline boards/wemos-zero/doc.txt:16]:
```suggestion
      Metro M0 Express should also work with this board definition.
```

?

File: boards/common/arduino-zero/include/arduino_iomap.h [hunk 3]
```diff
@@ -64,11 +64,26 @@ extern "C" {
  * @name    Mapping of MCU pins to Arduino pins
  * @{
  */
+
+/* Arduino Zero clones have D2 and D4 swapped compared to official boards, so
+ * we have handle them differently here. */
+#ifdef BOARD_WEMOS_ZERO
```
Comment (mguetschow) [inline boards/common/arduino-zero/include/arduino_iomap.h:70]: Is Wemos Zero the only Arduino Zero clone that we support?

**Review by mguetschow — CHANGES_REQUESTED** [review]
Thanks for fixing! Some nits below.

File: boards/wemos-zero/doc.txt (hunk 2 above)
Comment (maribu) [reply boards/wemos-zero/doc.txt]: Done and squashed

File: boards/common/arduino-zero/include/arduino_iomap.h (hunk 3 above)
Comment (mguetschow) [reply boards/common/arduino-zero/include/arduino_iomap.h:70]: Then maybe change the comment from "Arduino Zero clones have" to "Wemos Zero has"?

Comment (mguetschow) [reply boards/common/arduino-zero/include/arduino_iomap.h:70]: or "Wemos Zero boards have"

Comment (maribu) [reply boards/common/arduino-zero/include/arduino_iomap.h:70]: I rephrased the comment, fixed some sloppy copy-paste error, and squashed right away.

**Review by mguetschow — APPROVED** [review]
Thanks again!

**Comment by maribu** [comment]
Thx for the review :)

**Pull request merged**